presto-main/src/main/java/com/facebook/presto/operator/index/TupleFilterProcessor.java

```diff
@@ -71,6 +71,38 @@ public int process(ConnectorSession session, Page page, int start, int end, Page
         return end;
     }
 
+    @Override
+    public Page processColumnar(ConnectorSession session, Page page, List<? extends Type> types)
+    {
+        PageBuilder pageBuilder = new PageBuilder(types);
+        int positionCount = page.getPositionCount();
+
+        int[] selectedPositions = new int[positionCount];
+        int selectedCount = 0;
+        for (int i = 0; i < positionCount; i++) {
+            if (matches(i, page)) {
+                selectedPositions[selectedCount++] = i;
+            }
+        }
+
+        for (int position : selectedPositions) {
```

Comment:
The for loops should be reversed for columnar processing (and column variables pulled up)